### PR TITLE
expose av_seek_frame

### PR DESCRIPTION
**Pull request by stepancar**
Hello! Thank you for this project. 

I'm replicating functionality of this project https://github.com/Streampunk/beamcoder in browser.
this lib uses av_seek_frame here https://github.com/Streampunk/beamcoder/blob/master/src/demux.cc#L345
which works a little bit different from avformat_seek_file
I have exported it in my own custom build, but I wanted to ask is any chance to get it in official build?

### CODE REVIEW

**Comment by Yahweasel** [comment]
I have no opposition to exposing `av_seek_frame`. I originally only exposed `avformat_seek_file` because it's the new API that's supposed to deprecate the old `av_seek_frame` API, but it's been the new API that's supposed to deprecate the old API for at least five years, so who knows whether that will ever happen :)

However, a type declaration is needed. The reason why `avformat_seek_*` have `notypes` set is that they take int64s, and the way Emscripten passes int64s, that means its one parameter is actually two (lo bits and hi bits). See https://github.com/Yahweasel/libav.js/blob/cc3f81371629f5f833383be8ebaf6248aca022a0/libav.types.in.d.ts#L574

**Comment by stepancar** [comment]
> I have no opposition to exposing `av_seek_frame`. I originally only exposed `avformat_seek_file` because it's the new API that's supposed to deprecate the old `av_seek_frame` API, but it's been the new API that's supposed to deprecate the old API for at least five years, so who knows whether that will ever happen :)
> 
> However, a type declaration is needed. The reason why `avformat_seek_*` have `notypes` set is that they take int64s, and the way Emscripten passes int64s, that means its one parameter is actually two (lo bits and hi bits). See
> 
> https://github.com/Yahweasel/libav.js/blob/cc3f81371629f5f833383be8ebaf6248aca022a0/libav.types.in.d.ts#L574

Interesting. How does this test work then?
https://github.com/Yahweasel/libav.js/blob/cc3f81371629f5f833383be8ebaf6248aca022a0/tests/tests/615-seek.js#L37-L41

**Comment by Yahweasel** [comment]
wut

Well, one of these places is wrong. Will investigate.

**Comment by Yahweasel** [comment]
The test is definitely incorrect. Emscripten's documentation is clear—as clear as Emscripten's documentation gets, anyway—that int64 parameters are sent in pairs. What I'm now uncertain of is whether the type specification in funcs.json is *also* wrong. And, I'm also unsure why this utterly incorrect test is producing seemingly correct results.

**Comment by stepancar** [comment]
@Yahweasel , I have added ts definitions and test

File: libav.types.in.d.ts [hunk 1]
```diff
@@ -604,6 +604,14 @@ export interface LibAV extends LibAVStatic {
         flags: number
     ): Promise<number>;
 
+    /**
+     * Seek to the keyframe at timestamp.
+     */
+    av_seek_frame(
+        s: number, stream_index: number, tslo: number, tshi: number,
```
[the remark this answers is not in the document]
Comment (Yahweasel) [reply libav.types.in.d.ts:611]: "allow" is a strong word ;) . The functions exposed by libav.js don't *have* to be exactly those exposed by Emscripten; I just choose for them to be in most cases to have fewer layers of abstraction. This API could've been supported if, when I first added these int64 functions, I decided it was important enough.

**Comment by Yahweasel** [comment]
Thanks!

**Pull request merged**